### PR TITLE
TriggerAreaComponent(by boocho) + small fix

### PR DESCRIPTION
**Pull request by EternalWind**

### CODE REVIEW

File: engine/src/Logic/TriggerAreaComponent.hpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+#include <memory>
+
+#include <Config.hpp>
+
+#include <Scene/Component.hpp>
+#include <Logic/ScriptComponent.hpp>
+
+#include <btBulletCollisionCommon.h>
+#include <BulletCollision\CollisionDispatch\btGhostObject.h>
```
Comment (svenstaro) [inline engine/src/Logic/TriggerAreaComponent.hpp]: use /

File: engine/src/Logic/TriggerAreaComponent.cpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    if(!isEnabled()) {
+        return;
+    }
+
+    // TODO: Update only on node position change
+    btTransform translation;
+    translation.setOrigin(BtOgre::Convert::toBullet(getNode()->getPosition()));
+    mObject->setWorldTransform(translation);
+
+    for(uint32_t i = 0; i < mObject->getNumOverlappingObjects(); ++i)
```
Comment (svenstaro) [inline engine/src/Logic/TriggerAreaComponent.cpp]: You are comparing signed and unsigned here, don't do that.

**Comment by svenstaro** [comment]
Sample is crashing: 

INFO: Initializing input system (Window: 195035138)
INFO: Creating a scene manager for scene testscene.
DEBUG: Scene testscene is being initialized.
INFO: Initializing phyics world: testscene

Program received signal SIGSEGV, Segmentation fault.
0x00007ffff7af0475 in btMatrix3x3::operator= (this=0x8, other=...)
    at /usr/include/bullet/LinearMath/btMatrix3x3.h:70
70                      m_el[0] = other.m_el[0];
(gdb) bt
#0  0x00007ffff7af0475 in btMatrix3x3::operator= (this=0x8, other=...)
    at /usr/include/bullet/LinearMath/btMatrix3x3.h:70
#1  0x00007ffff7af1269 in btTransform::operator= (this=0x8, other=...)
    at /usr/include/bullet/LinearMath/btTransform.h:71
#2  0x00007ffff7b2691d in btCollisionObject::setWorldTransform (this=0x0, worldTrans=...)
    at /usr/include/bullet/BulletCollision/CollisionDispatch/btCollisionObject.h:297
#3  0x00007ffff7b265bf in dt::TriggerAreaComponent::onUpdate (this=0xb01690, time_diff=0)
    at /home/svenstaro/prj/ducttape-engine/engine/src/Logic/TriggerAreaComponent.cpp:22
#4  0x00007ffff7ace100 in dt::Node::_updateAllComponents (this=0xb01920, time_diff=0)
    at /home/svenstaro/prj/ducttape-engine/engine/src/Scene/Node.cpp:313
#5  0x000000000045b0ef in dt::Node::addComponentdt::TriggerAreaComponent (this=0xb01920, 
    component=0xb01690) at /home/svenstaro/prj/ducttape-engine/engine/src/Scene/Node.hpp:110
#6  0x000000000045a7bd in TriggerAreaComponentTest::Main::onInitialize (this=0x71aad0)
    at /home/svenstaro/prj/ducttape-engine/tests/src/TriggerAreaComponentTest/TriggerAreaComponentTest.cpp:83
#7  0x00007ffff7adb53f in dt::State::initialize (this=0x71aad0)
    at /home/svenstaro/prj/ducttape-engine/engine/src/Scene/State.cpp:19
#8  0x00007ffff7ad7dc6 in dt::StateManager::shiftStates (this=0x71a7a0)
    at /home/svenstaro/prj/ducttape-engine/engine/src/Scene/StateManager.cpp:59
#9  0x00007ffff7ad95e5 in dt::Game::run (this=0x7fffffffe060, start_state=0x71aad0, argc=2, 
    argv=0x7fffffffe318) at /home/svenstaro/prj/ducttape-engine/engine/src/Scene/Game.cpp:57
#10 0x0000000000459db7 in TriggerAreaComponentTest::TriggerAreaComponentTest::run (this=0x6eff70, argc=2, 
    argv=0x7fffffffe318)
    at /home/svenstaro/prj/ducttape-engine/tests/src/TriggerAreaComponentTest/TriggerAreaComponentTest.cpp:22
#11 0x0000000000469ec2 in main (argc=2, argv=0x7fffffffe318)
    at /home/svenstaro/prj/ducttape-engine/tests/src/TestFramework.cpp:120

**Comment by EternalWind** [comment]
Why does it crash? I only add 2 #include. Btw, did you test it before my changes? I even can't build it before that.

**Comment by svenstaro** [comment]
Crashes due to segfault. Last position in our engine is here:

#3  0x00007ffff7b275af in dt::TriggerAreaComponent::onUpdate (this=0xb01690, time_diff=0)
    at /home/svenstaro/prj/ducttape-engine/engine/src/Logic/TriggerAreaComponent.cpp:22

I checked the call and data at that point and it seems fine. The only thing that could be is that the data of mObject is corrupted. You should check into that.

**Pull request merged**